### PR TITLE
update closure-compiler to v20180319 and v20180402

### PR DESCRIPTION
**Pull request by teppeis**
- v20180204
    - fix some incorrect results
- v20180319
    - improve generators
- v20180402
    - improve Array#includes
    - add Promise#finally

cf. https://github.com/google/closure-compiler/wiki/Releases

### CODE REVIEW

File: environments.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     ]
   },
   "closure": {
-    "full": "Closure Compiler v20180204",
```
Comment (ljharb) [inline environments.json]: is there no value in keeping results from older versions?

Comment (teppeis) [reply environments.json]: @ljharb Closure Compiler is updated once every month or two month.
If this project can store the all releases, it's great and I would update this PR.

Comment (ljharb) [reply environments.json]: That seems ideal to me (assuming the old ones can be marked as obsolete/very as appropriate, so the UI stays manageable) - let's see what other collabs think.

Comment (teppeis) [reply environments.json]: This is separated column version: https://github.com/teppeis/compat-table/tree/closure-compiler-v20180319-new-column

**Review by ljharb — APPROVED** [review]

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**